client/src/containers/AssetAdmin/AssetAdmin.js

```diff
@@ -370,6 +367,10 @@ class AssetAdmin extends Component {
           this.handleBrowse(response.record.parent.id, response.record.id, null);
         }
         return response;
+      }).finally(() => {
```

Comment:
```suggestion
      }).then(() => {
```
Any reason why we're using finally() instead of regular then()?

Presumably if there's a hard failure in the previous then() block things are already in a broken so ensure the refetch happens probably won't fix things a whole lot

Reply:
Before this PR, it was _always_ refetching. Using `finally` ensures that's still the case - it will still _always_ refetch. This PR just changes _when_ it refetches (i.e. after the other request has fully completed)

Reply:
Yeah I get that, it just looks weird having a finally(), it implies we expect something to fail in the above catch() block.  It just makes me read in to it more then I should.

I think it's cleaner to just have a regular then() here instead

Reply:
I honestly don't know whether we need to always fetch or not - I wanted to keep that the same so I don't have to look into the failure states and whether refetching is necessary or not. My assumption is that the fact it always refetched before means there is a reason why we should always refetch.

Reply:
Will use `then()` for now. Fingers crossed there's not an edge case this breaks 😬